app/controllers/security_rulesets_controller.rb

```diff
@@ -25,7 +25,7 @@ def crunch42_data(upload)
     return File.read("fakecrunchresults.json") if ENV["BYPASS_API"] == "true"
 
     Tempfile.open(upload.oas_file.key) do |oas_file|
-      oas_file.write(upload.oas_file.download)
+      oas_file.write(upload.oas_file.download.encode('utf-8', 'binary', invalid: :replace, undef: :replace))
```

Comment:
[suggestion]

I'd be tempted to split that over a couple of lines. There is a lot going on there and if anything goes wrong it will be a little easier to unpick if the download is separated from the file write. So I'd suggest:
```ruby
content = upload.oas_file.download.encode('utf-8', 'binary', invalid: :replace, undef: :replace)
oas_file.write content
```